src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -51,6 +58,50 @@ internal partial class DiagnosticIncrementalAnalyzer
             {
                 throw ExceptionUtilities.Unreachable;
             }
+
+            static bool IsAnalyzerSuppressedForDocument(
+                DiagnosticAnalyzer analyzer,
+                BackgroundAnalysisScope analysisScope,
+                bool isActiveDocument,
+                bool isOpenDocument,
+                bool isGeneratedRazorDocument)
+            {
+                if (isGeneratedRazorDocument)
+                {
+                    // This is a generated Razor document, and they always want all analyzer diagnostics.
+                    return false;
+                }
+
+                bool analyzerEnabled;
+                if (analyzer.IsCompilerAnalyzer())
+                {
+                    // Compiler analyzer is treated specially.
+                    // It is executed for all documents (open and closed) for 'BackgroundAnalysisScope.FullSolution'
+                    // and executed for just open documents for other analysis scopes.
+                    analyzerEnabled = analysisScope == BackgroundAnalysisScope.FullSolution ? true : isOpenDocument;
```

Comment:
```suggestion
                    return analysisScope == BackgroundAnalysisScope.FullSolution || isOpenDocument;
```

Reply:
I changed this to `analyzerEnabled = analysisScope == BackgroundAnalysisScope.FullSolution || isOpenDocument;`. We want this method to return if analyzer is disabled, not enabled. The return statement at the end of the method returns `!analyzerEnabled`